dissect/target/tools/query.py

```diff
@@ -179,6 +179,30 @@ def main():
     for func in funcs:
         output_types.add(func.output_type)
 
+    # Verify if all provided plugins exist. Exits if a
+    # given function was not found in `find_plugin_functions`.
+    # TODO: Are there any other ways to call a function using target-query?
+    invalid_funcs = [func.strip(" *!?[]") for func in args.function.split(",")]
```

Comment:
This will break somewhat more complex patterns like: `browsers.*.downloads` (i.e. give me all downloads)

Reply:
Added handling for patterns such as `a.*.b` in commit [e3e4dbd](https://github.com/fox-it/dissect.target/pull/263/commits/e3e4dbd68c29187a406d1e1af37febcfdb749cf5). 

Are there any other ways to call a function using target-query?

Reply:
Yes, many unfortunately because the plugin finder uses fnmatch. It might be a better approach to use a return parameter in find_plugin_functions(... missing_plugins=[]) and check that?

Reply:
Thanks for the pointer. Implemented your suggestion in af13998f7255e1dfbd948afa09ec008f7632c0f2.